doc/jacoco.md

```diff
@@ -0,0 +1,164 @@
+      </execution>
+    </executions>
+    </plugin>
+   </plugins>
+  </build>
+</profile>
+```
+
+By default the generated report will be saved under `target/site/jacoco/jacoco.xml`; this location will be checked automatically by the sonar-jacoco plugin so no further configuration is required. Just launch `mvn sonar:sonar` as usual and the report will be picked up.
+
+If you need to change the directory where the report has been generated you can set the property either on the command line using maven's  `-D` switch 
+
+```
+mvn -Dsonar.coverage.jacoco.xmlReportPaths=report1.xml,report2.xml sonar:sonar 
+```
+
+or you can set the property inside your `pom.xml` 
+
+```xml
+    <properties>
+        <sonar.coverage.jacoco.xmlReportPaths>../app-it/target/site/jacoco-aggregate/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
+    </properties>
+
+```
+
+### Multi-module builds
+
+With multi-module builds, we sometimes need to show coverage across modules. For example, we might have multiple modules implementing business logic and another module which contains integration tests for all these modules. We would like to see also coverage from this integration test module on business logic modules.
+
+To achieve this, we can use JaCoCo's [`report-aggregate`](https://www.jacoco.org/jacoco/trunk/doc/report-aggregate-mojo.html) goal, which will collect coverage information from all modules and create a single report with coverage for the whole project. 
+
+First, we still have to use `prepare-agent` goal as we did in basic example in all modules, the best way to achieve this is to configure the execution of this goal in the parent `pom.xml`. Then we need to add execution to the module containing integration tests to generate the report across modules
+
+```
+<build>
+  <plugins>
+    <plugin>
+      <groupId>org.jacoco</groupId>
+      <artifactId>jacoco-maven-plugin</artifactId>
+      <executions>
+        <execution>
+          <id>report</id>
+          <goals>
+            <goal>report-aggregate</goal>
+          </goals>
+          <phase>verify</phase>
+        </execution>
+      </executions>
+    </plugin>
+  </plugins>
+</build>
+```
+
+We snippet binds the goal to the `verify` phase. Now when we execute
+
+```
+mvn clean verify
+```
+
+we should find the aggregated report in `target/site/jacoco-aggregate/jacoco.xml` of the module containing the integration tests.
+
+However, the JaCoCo plugin imports coverage reports module by module, so we need to import the same report multiple times (once for every module) to have coverage for all modules. To achieve this we set the property `sonar.coverage.jacoco.xmlReportPaths` in every module 
+
+```
+<properties>
+  <sonar.coverage.jacoco.xmlReportPaths>${project.basedir}/../integration_tests/target/site/jacoco-aggregate/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
+</properties>
+```
+
+We can factor out the path to the report in the property of the parent `pom.xml` to avoid repetition.
```

Comment:
I don't understand this.

Reply:
I removed it, it's not really necessary, and can be a bit confusing